File: Stack/stack.swift
import Foundation

protocol Stack {
    associatedtype Item

    var isEmpty: Bool { get }
    var isFull: Bool { get }
    var top: Int { get }
    var data: [Item] { get }

    func push(item: Item)
    func pop() -> Item?
}

class StackImpl<Item>: Stack {
    var isEmpty: Bool {
        data.isEmpty
    }

    var isFull: Bool {
        data.count == maxSize
    }

    var top: Int = -1

    var data: [Item] = []

    private var maxSize: Int

    init(maxSize: Int) {
        self.maxSize = maxSize
    }

    func push(item: Item) {
        guard !isFull else { return }
        top += 1
        data.append(item)
    }

    @discardableResult
    func pop() -> Item? {
        guard !isEmpty else { return nil }
        top -= 1
        return data.removeLast()
    }
}

let stack = StackImpl<Int>(maxSize: 10)

stack.pop()
stack.push(item: 3)
print(stack.data)
stack.push(item: 5)
print(stack.data)
stack.pop()
print(stack.data)
stack.push(item: 6)
print(stack.data)

File: Stack/stack.py
stack = []
max_size = 10


def isFull(stack):
    return len(stack) == max_size


def isEmpty(stack):
    return len(stack) == 0


def push(stack, item):
    if isFull(stack):
        print("스택이 가득 찼어요!")
    else:
        stack.append(item)
        print("데이터가 추가되었어요!", item)


def pop(stack):
    if isEmpty(stack):
        print("스택이 비었어요!")
        return None
    else:
        _item = stack.pop()
        print("데이터가 제거되었어요!", _item)
        return _item


push(stack=stack, item=3)
print(stack)
push(stack=stack, item=5)
print(stack)
pop(stack=stack)
print(stack)
push(stack=stack, item=6)
print(stack)
